---
title: "Class 14"
output: github_document
---

```{r}
counts <- read.csv("data/airway_scaledcounts.csv", stringsAsFactors = FALSE)
```

```{r}
metadata <-  read.csv("data/airway_metadata.csv", stringsAsFactors = FALSE)
```

```{r}
head(counts)
nrow(counts)
```

```{r}
head(metadata)
```

```{r}
nrow(metadata)
```

CHeck the col and row names
```{r}
colnames(counts) == metadata$id
```

##Section 3. Walk through of gene expression analysis

Metadata file tells us which columns of the count data are drug treated and control(no drug)

```{r}
control.inds <- metadata$dex == "control"
control <- metadata[control.inds,]
control
```

```{r}
head(counts[,control$id])
```


Determine the mean count values for all genes across control experiments

```{r}
control.mean <- rowSums(counts[,control$id])/length(control$id)
```

Do the same for the drug treated experiments

```{r}
treated.inds <- metadata$dex == "treated"
treated <- metadata[treated.inds,]
treated
```

```{r}
treated.mean <- rowSums(counts[,treated$id])/length(treated$id)
```


Store control and treated together for convenience

```{r}
meancounts <- data.frame(control.mean, treated.mean)
```

What is the sum counts accross all genes in control and treated

```{r}
colSums(meancounts)
```

```{r}
plot(meancounts)
```


Plot on log-log scale so we can see our data better
```{r}
plot(meancounts, log = "xy")
```

```{r}
meancounts$log2fc <- log2(meancounts[,"treated.mean"]/meancounts[,"control.mean"])
head(meancounts)
```

We will now remove the NaN and -Inf values
They come from zeros in the data. 



```{r}
to.rm <- unique(which(meancounts[,1:2] == 0, arr.ind=TRUE) [,1])
```
This line is complicated


```{r}
mycounts <- meancounts[-to.rm,]
nrow(mycounts)
```

How many genes are up in the drug treated cells?

```{r}
up.inds <- mycounts$log2fc > 2
sum(up.inds)
```

how many are down?


```{r}
down.inds <- mycounts$log2fc < -2
sum(down.inds)
```

##Section 4. Adding annotation data
We can add annotation from a supplied CSV file, such as those available from ENSEMBLE or UCSC. The annotables_grch38.csv annotation table links the unambiguous Ensembl gene ID to other useful annotation like the gene symbol, full gene name, location, Entrez gene ID, etc.

```{r}
anno <- read.csv("data/annotables_grch38.csv")
head(anno)
```



use the merge function to add the annotation data from the 'anno' object to our RNA-Seq results in 'mycounts'
```{r}
# use the merge 
mycounts.anno <- merge(mycounts, anno, by.x="row.names", by.y= "ensgene")
View(mycounts.anno)
```

Install these bioconductor annotation

```{r}
#BiocManager::install("AnnotationDbi")
#BiocManager::install("org.Hs.eg.db")
library("AnnotationDbi")
library("org.Hs.eg.db")
```

```{r}
columns(org.Hs.eg.db)
```

```{r}
head(mycounts)
```


We will use the **mapIds()** function from the bioconductor to add annotation data to our results
```{r}
mycounts$symbol <- mapIds(org.Hs.eg.db, 
       keys=row.names(mycounts),  # The gene neames in our dataset
      keytype="ENSEMBL",         # Format of our gene names
       column="SYMBOL")           # What new data do we want to add
```

```{r}
head(mycounts)
```

```{r}
mycounts$entrez <- mapIds(org.Hs.eg.db, 
       keys=row.names(mycounts),  # The gene neames in our dataset
       keytype="ENSEMBL",         # Format of our gene names
       column="ENTREZID")   
```

```{r}
head(mycounts)
```

##Section 5. Use DESeq2
```{r}
library("DESeq2")
```


```{r}
dds <- DESeqDataSetFromMatrix(countData=counts, 
                              colData=metadata, 
                              design=~dex, 
                              tidy=TRUE)
dds
```

```{r}
dds <-  DESeq(dds)
```

```{r}
res <- results(dds)
head(res)
```
we now have log2foldchange and pvalue

```{r}
summary(res)
```

```{r}
res05 <- results(dds, alpha = 0.05)
summary(res05)
```

store our results as a data.frame object
```{r}
res01 <- as.data.frame(results(dds, alpha=0.01))
head(res01)
```

Let's order our results by the adjusted p-value

```{r}
ord <- order( res01$padj )
#View(res01[ord.inds,])
head(res01[ord,])
```

```{r}
write.csv(res01[ord,], "signif01_results.csv")
```

##Volcano plot
```{r}
mycols <- rep("gray", nrow(res01))
mycols[res01$padj < 0.01 ] <- "black"
mycols[ res01$padj < 0.01 & abs(res01$log2FoldChange) > 2] <- "blue"
plot(res01$log2FoldChange, -log(res01$padj), xlab = "foldchange (Drug/Control)", ylab = "-log(P-value)",
     col=mycols)
```




```{r}
# Setup our custom point color vector 
mycols <- rep("gray", nrow(res))
mycols[ abs(res01$log2FoldChange) > 2 ]  <- "red" 


inds <- (res$padj < 0.01) & (abs(res$log2FoldChange) > 2 )
mycols[ inds ] <- "blue"
```


```{r}
#Volcano plot with custom colors
plot( res01$log2FoldChange, -log(res01$padj),
col=mycols, ylab="-Log(P-value)", xlab="Log2(FoldChange)" )

abline(v=c(-2,2), col="gray", lty=2)
abline(h=-log(0.1), col="gray", lty=2)
```
